[Teo Yu Qi] iP

teoyuqi opened the pull request:
# DukePro
> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))

DukePro frees your mind of having to remember things you need to do. It's,

* text-based
* easy to learn
* ~~FAST~~ SUPER FAST to use

All you need to do is,
1. download it from [here](https://github.com/teoyuqi/ip/releases/tag/A-Jar).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 

And it is **FREE**! :clown_face:

Features:

* [x] Managing tasks
* [x] Managing deadlines
* [ ] Reminders (coming soon)

---
If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:

```java
public class Duke {
    public static void main(String[] args) {
        new Duke(OUTPUT_DIRECTORY, OUTPUT_FILENAME).run();
    }
}
```

Commit:
Add Gradle support

teoyuqi committed:
Implement skeleton version of Duke

teoyuqi committed:
Add List attribute

teoyuqi committed:
Make dukeList static

teoyuqi committed:
Create DukeList data structure

teoyuqi committed:
Add javadocs

teoyuqi committed:
Add ability to store and list inputs

teoyuqi committed:
Rename DukeList to TaskList

teoyuqi committed:
Create Task object

teoyuqi committed:
Override toString

teoyuqi committed:
Add ability to mark tasks as done

teoyuqi committed:
Override toString for TaskList

teoyuqi committed:
Abstract out input handling

teoyuqi committed:
Create ToDo object

teoyuqi committed:
Create Event object

teoyuqi committed:
Create Deadline object

teoyuqi committed:
Remove echo function

teoyuqi committed:
Add code to print task when added

teoyuqi committed:
Add task types to functionality

teoyuqi committed:
Update Duke class path

teoyuqi committed:
Add hasNextLine check for Scanner

teoyuqi committed:
Add new line after each output

teoyuqi committed:
Remove extra whitespace

teoyuqi committed:
Add new test cases and expected ouput

teoyuqi committed:
Remove ability to add task without command

teoyuqi committed:
Add example from textbook

teoyuqi committed:
Create DukeException class

teoyuqi committed:
Add test for bye

teoyuqi committed:
Add first-class-citizen functions to handle commands

teoyuqi committed:
Add exception handling

teoyuqi committed:
Add test cases for exceptions

teoyuqi committed:
Replace addTask lambda with method reference

teoyuqi committed:
Add support for deleting tasks

teoyuqi committed:
Replace if-else with switch in add

teoyuqi committed:
Add test cases for deleting tasks

teoyuqi committed:
Use getSize method to get taskList size

teoyuqi committed:
Add ability to save tasks to disk

teoyuqi committed:
Fix bug deleting wrong index

teoyuqi committed:
Fix bug deleting wrong index

teoyuqi committed:
Update expected text for tests

teoyuqi committed:
Update expected text for tests

teoyuqi committed:
Add date parsing to `Deadline` objects

teoyuqi committed:
Add test for date parsing

teoyuqi committed:
Store date in Deadline as LocalDate object

teoyuqi committed:
Modify expected output of tests

teoyuqi committed:
Change deadline format when printing

teoyuqi committed:
Update expected outputs

teoyuqi committed:
Add method to create format for saving

teoyuqi committed:
Add `toFileFormat` method

teoyuqi committed:
Add 	oFileFormat method

teoyuqi committed:
Call parent `toFileFormat` method

teoyuqi committed:
Call parent 	oFileFormat method

teoyuqi committed:
Add 	oFile method

teoyuqi committed:
Add method to convert `taskList` to file format

teoyuqi committed:
Add new constructor

teoyuqi committed:
Add new constructor

teoyuqi committed:
Add method to add `Task` from file

teoyuqi committed:
Add method to add `Task` from file

teoyuqi committed:
Add ability to read tasks from disk

teoyuqi committed:
Add ability to read tasks from disk

teoyuqi committed:
Delete old tasklist on disk before test

teoyuqi committed:
Delete old tasklist on disk before test

teoyuqi committed:
Fix missing space in delimiter

teoyuqi committed:
Fix missing space in delimiter

teoyuqi committed:
Fix merge conflicts

teoyuqi committed:
Fix merge conflicts

teoyuqi committed:
Merge `branch-level-8` to `master`

teoyuqi committed:
Rebase to fix typo in old commit message

teoyuqi committed:
Merge branch remote `branch-level-7' into `branch-level-7`

teoyuqi committed:
Merge `branch-level-7` into `master`

teoyuqi committed:
Extract logo and command info to Ui object

teoyuqi committed:
Fix bug in parsing dates from file

teoyuqi committed:
Extract file reading and writing to `Storage`

teoyuqi committed:
Remove file io modules

teoyuqi committed:
Remove println used for debugging

teoyuqi committed:
Create Parser object

teoyuqi committed:
Extract parsing to Parser object

teoyuqi committed:
Extract println to Ui object

teoyuqi committed:
Create 
un method for Duke

teoyuqi committed:
Move files to packages

teoyuqi committed:
Modify class file path

teoyuqi committed:
Modify expected text for test script

teoyuqi committed:
Merge branch 'add-gradle-support' of github.com:nus-cs2103-AY2223S1/ip

teoyuqi committed:
Fix package naming

teoyuqi committed:
Change java file path in Gradle build file

teoyuqi committed:
Add saved task list to .gitignore

teoyuqi committed:
Add JUnit test for Parser object

teoyuqi committed:
Add JUnit test for Event object

teoyuqi committed:
Add JavaDocs for `Duke`

teoyuqi committed:
Edit JavaDocs for `Duke`

teoyuqi committed:
Add JavaDocs for `DukeException`

teoyuqi committed:
Add JavaDocs for `Parser`

teoyuqi committed:
Add JavaDocs to `Storage`

teoyuqi committed:
Restyle JavaDocs

teoyuqi committed:
Restyle JavaDocs

teoyuqi committed:
Add JavaDocs to `TaskList`

teoyuqi committed:
Add JavaDocs to `Ui`

teoyuqi committed:
Add JavaDocs to `Deadline`

teoyuqi committed:
Edit JavaDocs

teoyuqi committed:
Add JavaDocs to `Event`

teoyuqi committed:
Edit JavaDocs

teoyuqi committed:
Add JavaDocs to `Task`

teoyuqi committed:
Add JavaDocs to `ToDo`

teoyuqi committed:
Follow CS2103T coding standard

teoyuqi committed:
Add method to find word in `Task` description

teoyuqi committed:
Add method to find word in `Deadline` description and date

teoyuqi committed:
Add method to find word in `Event` description and date

teoyuqi committed:
Add method to find word among `Task`s in `TaskList`

teoyuqi committed:
Add command to find word in `TaskList`

teoyuqi committed:
Add `find` command in `AVAILABLE_COMMANDS`

teoyuqi committed:
Merge branch `branch-Level-7`

teoyuqi committed:
Merge branch `branch-A-JavaDoc`

teoyuqi committed:
Merge branch `branch-A-CodingStandard`

teoyuqi committed:
Merge branch `branch-Level-9`

teoyuqi committed:
Add CheckStyle

teoyuqi committed:
Edit JavaDocs style

teoyuqi committed:
Modify in accordance to CheckStyle

teoyuqi committed:
Rename `Ui` attribute name

teoyuqi committed:
Merge `branch-A-CheckStyle` into `Master`

teoyuqi committed:
Add `JavaFX` to dependencies

teoyuqi committed:
Change to full graphical user interface

teoyuqi committed:
Merge `branch-Level-10` into `master`

teoyuqi committed:
Fix bug which changed long texts in DialogBox into ellipsis

teoyuqi committed:
Merge `branch-Level-10` into `master` after bug fix

teoyuqi committed:
Change `AVAILABLE_COMMANDS` to include date format

teoyuqi committed:
Add `CSS` stylesheets to `FXML`

songivan00 commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,348 @@
+     *  </ol>
+     * </p>
+     */
+    private static final BiFunction<TaskList, String, String> mark = (
```
I don't think this is a big issue. Perhaps you can change the method naming to markTask, since you are using names like addTask and deleteTask too?

teoyuqi replied:
Thank you for pointing that out! I have rectified this in [commit f6a37d2](https://github.com/nus-cs2103-AY2223S1/ip/pull/15/commits/f6a37d2746efaaa72e4406874f531579b2db729b).

songivan00 commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,97 @@
+import java.io.IOException;
+import java.util.Scanner;
+
+/**
+ * Defines <Code>Storage</Code> class.
+ * <p>In charge of reading and writing files from disk.</p>
+ */
```
I like how you added javadocs to all your classes, fields and methods.

teoyuqi replied:
Thank you for your kind comments!

songivan00 commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,215 @@
+     * @param index Index of task to be undone. 1 based indexing.
+     * @return      <Code>String</Code> output to be shown to user.
+     */
+    public String markUnDone(int index) {
```
Should this be markUndone instead?

teoyuqi replied:
Thank you for pointing that out! `markUnDone` has been renamed to `markUndone` in [commit 0a111e6](https://github.com/nus-cs2103-AY2223S1/ip/pull/15/commits/0a111e6647fcb5bf6706da0f8a986790952f6319).

songivan00 commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,97 @@
+     * @param lineNumberInFile Line number of current line.
+     * @param taskList         <Code>TaskList</Code> object to add tasks to.
+     */
+    private static void addLineFromFile(String data, int lineNumberInFile,
```
I like how most of your method's naming follow the coding standard. Good job!!

teoyuqi replied:
Thank you for your kind comments! The issues you raised have been fixed. 😃

teoyuqi committed:
Rename `BiFunction`s `mark`, `unmark`, `list` and `find` for added clarity

teoyuqi committed:
Rename `markUnDone` to `markUndone` in line with coding standards

teoyuqi committed:
Rename `markUnDone` to `markUndone` in line with coding standards

teoyuqi committed:
Remove extra whitespace

teoyuqi committed:
Change `mainClassName` to `Launcher`

`mainClassName` is `duke.Duke`.

Since the addition of the GUI, the `main` function is now
located in `Launcher`. As such, the `mainClassName` should
be changed to `Launcher` to ensure that the application is
built correctly by `Gradle`.

Let's, change `mainClassName` from `dukeDuke` to `Launcher`.

teoyuqi committed:
`assert` that no known commands enter `default` case

teoyuqi committed:
Add assertions when adding new tasks

When the `add`, `addFromFile` and `getDescription` methods are called,
the input is assumed to have been a `deadline`, `event` or `todo`
command. Other commands are ignored (without warnings or exceptions).

Errorneous behaviour that results in calling `add`, `addFromFile` or
`getDescription` on inputs that are not `deadline`, `event` or `todo`
tasks can be picked up by adding assertions. This will aid in future
debugging.

Let's add an `assert` statement in the `default` and `else` blocks of
`add`, `addFromFile` and `getDescription.

teoyuqi committed:
Extract input validation from `handleUserInputs` to individual methods

teoyuqi committed:
Refactor to avoid complicated `Boolean` expressions

teoyuqi committed:
Refactor to avoid complicated `Boolean` expressions

teoyuqi committed:
Merge pull request #1 from teoyuqi/branch-A-CodeQuality

Merge `branch-A-CodeQuality` into `master`

teoyuqi committed:
Merge pull request #1 from teoyuqi/branch-A-CodeQuality

Merge `branch-A-CodeQuality` into `master`

teoyuqi committed:
Merge branch 'master' into branch-A-Assertions

teoyuqi committed:
Merge pull request #2 from teoyuqi/branch-A-Assertions

Merge `branch-A-Assertions` into `master`

teoyuqi committed:
Merge branch 'master' of github.com:teoyuqi/ip

teoyuqi committed:
Merge remote-tracking branch 'own-remote/master'

teoyuqi committed:
Use GitHub Actions to set up CI

teoyuqi committed:
Create `Enum` type to indicate priority levels

teoyuqi committed:
Add `Level` attribute to assign task's priority

teoyuqi committed:
Add `Level` parameter to constructor

teoyuqi committed:
Add priority level to file format

teoyuqi committed:
Add priority level when adding new tasks

teoyuqi committed:
Add priority level to string format

teoyuqi committed:
Account for priority in `getDate` and `getVenue`

teoyuqi committed:
Fix error in index of priority in input

teoyuqi committed:
Add ability to input priority

teoyuqi committed:
Add ability to input priority

teoyuqi committed:
Check that priority exists before adding line from file

teoyuqi committed:
Fix wrong `endIndex` bug

teoyuqi committed:
Change input to lower case before checking priority

teoyuqi committed:
Fix incorrect array index

teoyuqi committed:
Fix incorrect `endIndex` when parsing dates

teoyuqi committed:
Merge `branch-C-Priority` into `master`

teoyuqi committed:
Remove IO redirection test

teoyuqi committed:
Change variable naming according to `CheckStyle`

shamooose reviewed (commented):
Good upkeeping of coding standards!

teoyuqi committed:
Add priority argument to constructor

teoyuqi committed:
Update expected outputs with task priorities

teoyuqi committed:
Rename variable to be in line with coding standards

teoyuqi committed:
Allow resizing of windows

teoyuqi committed:
Change profile pictures to circular crop

teoyuqi committed:
Merge `branch-A-BetterGui` into `master`

teoyuqi committed:
Remove unused `scene` variable

teoyuqi committed:
Rename `Boolean` variable according to coding standard

teoyuqi committed:
Modify header comments

teoyuqi committed:
Remove border color

teoyuqi committed:
Change application logo to Duke's profile picture

teoyuqi committed:
Remove text wrapping in `Label`

teoyuqi committed:
Increase minimum height of `DialogBox`

teoyuqi committed:
Add screenshot of user interface

teoyuqi committed:
Update `README.md`

teoyuqi committed:
Update `README.md`

teoyuqi committed:
Center images

teoyuqi committed:
Merge `branch-A-UserGuide` into `master`

teoyuqi committed:
Update `README.md`

teoyuqi committed:
Change page theme

teoyuqi committed:
Fix bug when parsing `ToDo` tasks

teoyuqi committed:
Merge branch 'master' of github.com:teoyuqi/CS2103T-Individual-Project

teoyuqi committed:
Rename variable according to CS2103 coding standards
